app/src/main/java/ru/prometeydev/movie/data/adapters/MoviesAdapter.kt

```diff
@@ -60,29 +66,39 @@ class MoviesAdapter(
         private val like = itemView.findViewById<ImageView>(R.id.like_heart)
 
         fun onBind(movie: Movie, clickListener: OnRecyclerItemClicked) {
-            name.text = movie.name
-            genre.text = movie.genre
-            duration.text = context.getString(R.string.movie_time, movie.duration)
-            rating.rating = movie.rating
-            reviewsCount.text = context.getString(R.string.reviews, movie.reviewsCount)
-            ageLimit.text  = context.getString(R.string.age_limit, movie.ageLimit)
-            filmCover.setImageResource(movie.filmCoverDrawable)
-
-            like.setImageResource(
-                    if (movie.hasLike)
-                        R.drawable.ic_like_active
-                    else
-                        R.drawable.ic_like_inactive
-            )
+            name.text = movie.title
+            genre.text = movie.genres.joinToString { it.name }
+            duration.text = context.getString(R.string.movie_time, movie.runtime)
+            rating.rating = movie.ratings.calculateStarsCount()
+            reviewsCount.text = context.getString(R.string.reviews, movie.numberOfRatings)
+            ageLimit.text  = context.getString(R.string.age_limit, movie.minimumAge)
+
+            scope.launch {
+                filmCover.load(loadImage(movie.poster)) {
+                    crossfade(true)
+                    placeholder(R.drawable.ic_image_placeholder)
+                    fallback(R.drawable.ic_image_placeholder)
+                }
+            }
 
             itemView.setOnClickListener {
                 clickListener.onClick(movie)
             }
         }
 
+        private suspend fun loadImage(url: String): Drawable? = withContext(Dispatchers.IO) {
```

Comment:
Is this exercise since I saw something like this already in coil?

Reply:
I think a simple call to load(url) will be enough.